core/permissions.py

```diff
@@ -1,12 +1,27 @@
-from rest_framework import permissions
+from rest_framework import permissions, exceptions
+from authentication.models import User
 
 class HackathonPermissions(permissions.BasePermission):
-        '''
+        """
         1. GET can be accessed even by anonymous users
         2. PUT, PATCH and DELETE can only be accessed by the Super User
-        '''
+        """
         def has_permission(self, request, view):
             if request.method in permissions.SAFE_METHODS:
                 return True
             return bool(request.user and request.user.is_superuser)
-    
\ No newline at end of file
+
+class isProfileComplete(permissions.BasePermission):
+    """
+    1. GET can be accessed even by anonymous users
+    2. POST can only be accessed by users whose profile is completed
+    """
+    def has_permission(self, request, view):
+        if request.method == 'GET':
+            return True
```

Comment:
This should be a general permission. So, maybe remove this,

Reply:
done